app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Post;
+use Illuminate\Http\Request;
+use Illuminate\Support\Str;
+
+class PostController extends Controller
+{
+    public function index()
+    {
+        $posts = Post::orderBy('created_at', 'desc')->get();
+        return view('posts.index', compact('posts'));
+    }
+
+    public function show(Post $post)
+    {
+        return view('posts.show', [
+            'post' => $post
+        ]);
+    }
+
+    public function create()
+    {
+        return view('posts.create');
+    }
+
+    public function store(Request $request)
+    {
+        $validated = $request->validate([
+            'title' => 'required|string|min:10',
+            'summary' => 'required|string|min:10',
+            'body' => 'required|string|min:10'
+            ]);
+
+        $validated['slug'] = Str::slug($validated['title']);
+
+        $post = new Post($validated);
+
+        $post->save();
+
+        return redirect()->route('posts.index');
+    }
+
+    public function edit(Post $post)
+    {
+        return view('posts.edit', [
+            'post' => $post
+        ]);
+    }
+
+    public function update(Request $request, Post $post)
+    {
+        $request->validate([
+            'title' => 'required|string|min:10',
+            'summary' => 'required|string|min:10',
+            'body' => 'required|string|min:10'
+        ]);
+
+        $post->update($request->all());
```

Comment:
When the user updates the title, the slug is not updated now. It might be more convenient to implement the slug with an accessor and/or mutator,